nx/lib/nx/defn/expr.ex

```diff
@@ -816,7 +816,8 @@ defmodule Nx.Defn.Expr do
     end)
   end
 
-  defp tuple(%T{type: {:tuple, size}, data: %{context: context}} = expr, list)
+  @doc false
```

Comment:
Please move it to the top of this module (around `def metadata`) and add a public doc to it. I think it is a good practice to treat `grad` as a public consumer: i.e. it should not use any private API.

Reply:
Oh, ok. Any suggestions for the doc?

Reply:
I've moved this. There's also defp tuple_out. Should I leave it where it is?

Reply:
You can bring it close to tuple but also leave it as is, your call.